### PR TITLE
Bug fix in sandbox start when sandbox exist

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Bugfix in sandbox start when sandbox exists.  (Before fix if you say no on the existing cluster then ideally it should return exit code 0 but it will continue sandbox start process that was wrong)
- update upgrade in docs

```bash
$ flytectl sandbox start 
delete existing sandbox cluster [y/n]: n
```

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: cmd/sandbox/start_test.go [hunk 1]
```diff
@@ -57,6 +57,43 @@ func TestStartSandboxFunc(t *testing.T) {
...
+		mockDocker.OnImagePullMatch(ctx, mock.Anything, types.ImagePullOptions{}).Return(os.Stdin, nil)
+		mockDocker.OnContainerLogsMatch(ctx, mock.Anything, types.ContainerLogsOptions{
+			ShowStderr: true,
+			ShowStdout: true,
+			Timestamps: true,
+			Follow:     true,
+		}).Return(nil, nil)
+		mockDocker.OnContainerWaitMatch(ctx, mock.Anything, container.WaitConditionNotRunning).Return(bodyStatus, errCh)
+		_, err := startSandbox(ctx, mockDocker, strings.NewReader("n"))
+		assert.Nil(t, err)
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start_test.go:95]: assert also that the sandbox exists in the list of containers

File: pkg/docker/docker_util.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func RemoveSandbox(ctx context.Context, cli Docker, reader io.Reader) error {
 			})
 			return err
 		}
-		return nil
+		return fmt.Errorf("EXIST")
```
Comment (pmahindrakar-oss) [inline pkg/docker/docker_util.go:72]: Define this error as a constant and use it here aswell https://github.com/flyteorg/flytectl/pull/145/files#diff-81ed1d7a8cce59819e2ac00f20daaf9cc8e86219790a191ed190c8c6e22ffed8R83

And lets add a more descriptive error here rathere just EXIST.

Lets calls it SandboxExists

File: cmd/sandbox/start_test.go (hunk 1 above)
Comment (evalsocket) [reply cmd/sandbox/start_test.go:95]: yes, I did it...In mock I return the sandbox container list

File: cmd/sandbox/start.go [hunk 3]
```diff
@@ -68,9 +70,14 @@ func startSandboxCluster(ctx context.Context, args []string, cmdCtx cmdCore.Comm
 
 	reader, err := startSandbox(ctx, cli, os.Stdin)
 	if err != nil {
-		return err
+		if err.Error() != clierrors.ErrSandboxExists {
+			return err
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start.go:74]: With your previous change exit code was 0. This will cause a non-zero exit code . Is that what was desired in the first place

Comment (pmahindrakar-oss) [reply cmd/sandbox/start.go:74]: != . got it

**Review by pmahindrakar-oss — APPROVED** [review]

File: cmd/sandbox/start.go (hunk 3 above)
Comment (evalsocket) [reply cmd/sandbox/start.go:74]: yes

**Review by pmahindrakar-oss — APPROVED** [review]
